src/main/scala/is/hail/annotations/aggregators/RegionValueVirtualDSAggregator.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple}
+import org.apache.spark.sql.Row
+
+object RegionValueVirtualDSAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64())))
+}
+
+class RegionValueVirtualDSAggregator(xmin: Double, xmax: Double, ymin: Double, ymax: Double) extends RegionValueAggregator {
+  private var boxes = Set.empty[(Int, Int)]
+
+  def seqOp(r: Region, offset: Long, missing: Boolean) {
+    if (!missing) {
+      val row = UnsafeRow.read(TTuple(Array(TFloat64(), TFloat64())), r, offset).asInstanceOf[UnsafeRow]
+      val x = row.getDouble(0)
+      val y = row.getDouble(1)
+      val i = ((((x - xmin) / (xmax - xmin)) * 10).toInt, (((y - ymin) / (ymax - ymin)) * 10).toInt)
+      boxes += i
+    }
+  }
+
+  def combOp(agg2: RegionValueAggregator) {
+    boxes = boxes ++ agg2.asInstanceOf[RegionValueVirtualDSAggregator].boxes
+  }
+
+  def result(rvb: RegionValueBuilder) {
+    rvb.startArray(boxes.size)
+    boxes.iterator.foreach { i =>
+      val x = (i._1 / (xmax - xmin)) * 10
+      val y = (i._2 / (ymax - ymin)) * 10
```

Comment:
I don't think this is quite right.  I ranges (loosely) from 0 to N.  So if you divide it by N, it goes from 0 to 1.  Then if you mutliply by (ymax - ymin), to goes from 0 to (ymax - ymin).  Then you add ymin, and it goes from ymin to ymin + (ymax - ymin) = ymax, as desired.  So that gets you:

  x = xmin + ((i._1 + 0.5) / N * (xmax - xmin))

(where the 0.5 comes from the comment above).